zilliqa/src/api/types/zil.rs

```diff
@@ -618,6 +618,136 @@ pub struct TransactionReceiptResponse {
     pub success: bool,
 }
 
+#[derive(Serialize, Deserialize, Clone)]
+pub struct TransactionStatusResponse {
+    #[serde(rename = "ID")]
+    pub id: String,
+    #[serde(rename = "_id")]
+    pub _id: serde_json::Value,
+    pub amount: String,
+    pub data: String,
+    #[serde(rename = "epochInserted")]
+    pub epoch_inserted: String,
+    #[serde(rename = "epochUpdated")]
+    pub epoch_updated: String,
+    #[serde(rename = "gasLimit")]
+    pub gas_limit: String,
+    #[serde(rename = "gasPrice")]
+    pub gas_price: String,
+    #[serde(rename = "lastModified")]
+    pub last_modified: String,
+    #[serde(rename = "modificationState")]
+    pub modification_state: u64,
+    pub status: u64,
```

Comment:
It might be useful to link to https://github.com/Zilliqa/Zilliqa/blob/master/src/common/TxnStatus.h#L23 somewhere so we know what `1`, `3`, etc. mean.